pkg/datastore/filedb/filter.go

```diff
@@ -15,19 +15,181 @@
 package filedb
 
 import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
 	"github.com/pipe-cd/pipecd/pkg/datastore"
 )
 
-// TODO: Implement filterable interface for each collection.
 type filterable interface {
 	Match(e interface{}, filters []datastore.ListFilter) (bool, error)
 }
 
 func filter(col datastore.Collection, e interface{}, filters []datastore.ListFilter) (bool, error) {
+	// If the collection implement filterable interface, use it.
 	fcol, ok := col.(filterable)
-	if !ok {
+	if ok {
+		return fcol.Match(e, filters)
+	}
+
+	// remarshal entity as map[string]interface{} struct.
+	raw, err := json.Marshal(e)
+	if err != nil {
+		return false, err
+	}
+	var omap map[string]interface{}
+	if err := json.Unmarshal(raw, &omap); err != nil {
+		return false, err
+	}
+
+	for _, filter := range filters {
+		field := convertCamelToSnake(filter.Field)
+		if strings.Contains(field, ".") {
+			// TODO: Handle nested field name such as SyncState.Status.
+			return false, datastore.ErrUnsupported
```

Comment:
Idea: Or to be simple, we can avoid using nested fields by double storing the SyncState_Status in the Application model.

Reply:
I see, I planned to make logic like flatten nested json and do filter logic on the output map. But you're right, we may avoid complexity by unuse nested fields query. Will think about it and address with another PR 🙏